rules/prefer-at.js

```diff
@@ -96,9 +96,12 @@ function checkSliceCall(node) {
 	const startIndex = -startIndexNode.argument.value;
 	if (sliceArgumentsLength === 1) {
 		if (
-			firstElementGetMethod === 'zero-index'
-			|| firstElementGetMethod === 'shift'
-			|| (startIndex === -1 && firstElementGetMethod === 'pop')
+			startIndexNode.argument.value === 1
+			&& (
+				firstElementGetMethod === 'zero-index'
+				|| firstElementGetMethod === 'shift'
+				|| (startIndex === -1 && firstElementGetMethod === 'pop')
+			)
```

Comment:
This is incorrect, the value of `startIndexNode` doesn't matter. It only result differently when the index is out of the boundary of string. And sting don't have `.shift`/`.pop`, so it doesn't matter.

Reply:
@fisker we're checking `startIndexNode` because `.slice(-1)[0]` can still be safely converted to `.at(-1)`, but only for `-1`, not any other number. For any other number the behaviour is different if the index is out of bounds

Reply:
You are right.

But i think we should still report `array.slice(-9)[0]`, just remove the autofix.